### PR TITLE
Add padding support to wvSplitK solution for skinny GEMMs

### PR DESCRIPTION
**Pull request by amd-hhashemi**
Adds padding support to wvSplitK solution for skinny GEMMs

**MI355 Performance:**

vllm bench serve
model=atiiuae/**falcon-mamba-tiny-dev** [a model where padded skinny GEMMs occur)
max_concurrency=1
num_prompts=400
in=1024
out=1024

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/hhashemi/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/hhashemi/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

  | before | after | % change
-- | -- | -- | --
Total token throughput (tok/s) | 2839.31 | 2864.59 | 0.89%
Mean TTFT (ms) | 11.41 | 10.91 | -4.38%
Mean ITL (ms) | 0.76 | 0.76 | 0.00%
Mean E2EL (ms) | 721.08 | 714.73 | -0.88%

</body>

</html>

**Accuracy:**

branch-main:
local-completions ({'model': 'deepseek-r1-FP8-Dynamic', 'base_url': 'http://127.0.0.1:8000/v1/completions'}), gen_kwargs: ({}), limit: None, num_fewshot: None, batch_size: 4
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9500|±  | 0.006|
|     |       |strict-match    |     5|exact_match|↑  |0.9492|±  | 0.006|

branch-wvSplitKpd:
local-completions ({'model': 'deepseek-r1-FP8-Dynamic', 'base_url': 'http://127.0.0.1:8000/v1/completions'}), gen_kwargs: ({}), limit: None, num_fewshot: None, batch_size: 4
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9515|±  |0.0059|
|     |       |strict-match    |     5|exact_match|↑  |0.9507|±  |0.0060|

branch-main:
vllm ({'pretrained': 'openai/gpt-oss-120b', 'tensor_parallel_size': 1, 'dtype': 'auto', 'gpu_memory_utilization': 0.8}), gen_kwargs: ({}), limit: None, num_fewshot: None, batch_size: 4
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.4458|±  |0.0137|
|     |       |strict-match    |     5|exact_match|↑  |0.2707|±  |0.0122|

branch-wvSplitKpd:
vllm ({'pretrained': 'openai/gpt-oss-120b', 'tensor_parallel_size': 1, 'dtype': 'auto', 'gpu_memory_utilization': 0.8}), gen_kwargs: ({}), limit: None, num_fewshot: None, batch_size: 4
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.4556|±  |0.0137|
|     |       |strict-match    |     5|exact_match|↑  |0.2798|±  |0.0124|

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by AndreasKaratzas** [comment]
@amd-hhashemi Thank you for taking a look into it. The failures are from today's CI nightly run: https://buildkite.com/vllm/amd-ci/builds/4312/steps/canvas?jid=019c31c0-3f89-47e5-b814-f2b9f513c8d4&tab=output

This is a new regression, and it did not happen in the last nightly: https://buildkite.com/vllm/amd-ci/builds/4208
List of all latest nightlies: https://buildkite.com/vllm/amd-ci/builds?branch=main

**Comment by AndreasKaratzas** [comment]
@amd-hhashemi Understood, thank you for clarification. I'm also going to open an issue then on PyTorch. It's unclear to me why 5e-2 is acceptable.

**Comment by tjtanaa** [comment]
@amd-hhashemi Please show the test results for any bugfix PR so that we know the states of the test group. The tests group related to the test file you are fixing is not triggered in the AMD CI. We have to set `AMD_MIRROR_HW="amdexperimental"`

**Comment by amd-hhashemi** [comment]
> @amd-hhashemi Please show the test results for any bugfix PR so that we know the states of the test group. The tests group related to the test file you are fixing is not triggered in the AMD CI. We have to set `AMD_MIRROR_HW="amdexperimental"`

@tjtanaa  Will do. I've expanded the tests to cover all padding/bias scenarios. Offline tests are run with extended seed variations, Full mi325 run after this PR and #34013 are attached 
[test_rocm_skinny_gemms_MI355_befor.log](https://github.com/user-attachments/files/25168086/test_rocm_skinny_gemms_MI355_befor.log)
[test_rocm_skinny_gemms_mi325_after33762.log](https://github.com/user-attachments/files/25168864/test_rocm_skinny_gemms_mi325_after33762.log)

**Review by gshtras — APPROVED** [review]

**Review by tjtanaa — APPROVED** [review]
